server/src/main/java/org/elasticsearch/cluster/metadata/DataStream.java

```diff
@@ -73,7 +74,8 @@
 
     private final LongSupplier timeProvider;
     private final String name;
-    private final TimestampField timeStampField;
+    // This was always fixed to @timestamp with the idea that one day this field could be configurable. This idea no longer exists.
+    private final TimestampField timeStampField = TIMESTAMP_FIELD;
```

Comment:
Nitpick: If this is not going to be configurable, shouldn't we use the constant within the class and remove the field?

Reply:
Yes, that makes sense. The constant can be returned in the getter.